specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json

```diff
@@ -0,0 +1,646 @@
+          },
+          "201": {
+            "description": "Scan result was converted successfully.",
+            "schema": {
+              "$ref": "#/definitions/DatabaseVulnerabilityAssessmentScansExport"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Export a database's vulnerability assessment scan results.": {
+            "$ref": "./examples/DatabaseVulnerabilityAssessmentScanExport.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "description": "ARM resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {}
+    },
+    "Sku": {
+      "description": "An ARM Resource SKU.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the SKU, typically, a letter + Number code, e.g. P3.",
+          "type": "string"
+        },
+        "tier": {
+          "description": "The tier of the particular SKU, e.g. Basic, Premium.",
+          "type": "string"
+        },
+        "size": {
+          "description": "Size of the particular SKU",
+          "type": "string"
+        },
+        "family": {
+          "description": "If the service has different generations of hardware, for the same SKU, then that can be captured here.",
+          "type": "string"
+        },
+        "capacity": {
+          "format": "int32",
+          "description": "Capacity of the particular SKU.",
+          "type": "integer"
+        }
+      }
+    },
+    "DatabaseVulnerabilityAssessmentProperties": {
+      "description": "Properties of a database Vulnerability Assessment.",
+      "type": "object",
+      "properties": {
+        "storageContainerPath": {
+          "description": "A blob storage container path to hold the scan results (e.g. https://myStorage.blob.core.windows.net/VaScans/).",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ]
+        },
+        "storageContainerSasKey": {
+          "description": "A shared access signature (SAS Key) that has write access to the blob container specified in 'storageContainerPath' parameter.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ]
+        },
```

Comment:
The storage account should be referenced using the fully qualified ARM resource id instead of passing the url and access-key. You can then enable linked access check on it reliably to ensure sufficient permission are there to perform the operation. Since this is an ARM guideline, tagging Gaurav @ravbhatnagar

Reply:
We already got the approval from ARM to go this way before.
The intention here is that a user will supply a SAS key (not access key) with an expiration date and privileges he chooses to supply.
This way he is more "secure" as we have a SAS key that he controls on.